### DISCUSSION TITLE
Conversion of request objects to dict

### DISCUSSION THREAD

**Discussion by do-me** [encode/starlette#1551] [Ideas] [answered]
Hi everyone, 

## Idea 
I'm writing my own logging module and noticed that some of [starlette's `Request` objects](https://www.starlette.io/requests/#client-address) can be converted to a normal `dict` and others can't.

Works: `dict(request.headers)` -> `{'host': 'localhost:8000', 'user-agent': 'Mozilla/5.0 ...',...}`
Doesn't: `dict(request.client)` -> `*** ValueError: dictionary update sequence element #0 has length 9; 2 is required`

I know I can access the `Headers` or `Client` objects _like_ a dict, but as I'm logging it - and for the sake of **consistency** - it might be nice to enable easy `dict` conversion for all objects. 

The advantage is that I when I'm logging the normal `dict` (-> `{'host': 'localhost:8000', 'user-agent': 'Mozilla/5.0 ...',...`) there is no unnecessary string (e.g. `Headers(...)`) that would be included when directly logging the object (-> `Headers({'host': 'localhost:8000', 'user-agent': 'Mozilla/5.0`).

## Workaround

My current "workaround" is to return a tuple by indexing it like `request.client[:]` -> `('127.0.0.1', 12345)`.

**Comment by Kludex** [comment, marked as answer]
I think you want the `request.scope`.
